modules/light-clients/08-wasm/types/contract_api.go

```diff
@@ -66,21 +67,23 @@ type UpdateStateOnMisbehaviourMsg struct {
 
 // VerifyMembershipMsg is a sudoMsg sent to the contract to verify a membership proof.
 type VerifyMembershipMsg struct {
-	Height           clienttypes.Height         `json:"height"`
-	DelayTimePeriod  uint64                     `json:"delay_time_period"`
-	DelayBlockPeriod uint64                     `json:"delay_block_period"`
-	Proof            []byte                     `json:"proof"`
-	Path             commitmenttypes.MerklePath `json:"path"`
-	Value            []byte                     `json:"value"`
+	Height           clienttypes.Height            `json:"height"`
+	DelayTimePeriod  uint64                        `json:"delay_time_period"`
+	DelayBlockPeriod uint64                        `json:"delay_block_period"`
+	Proof            []byte                        `json:"proof"`
+	MerklePath       *commitmenttypesv2.MerklePath `json:"merkle_path,omitempty"`
+	Path             commitmenttypes.MerklePath    `json:"path"`
```

Comment:
Yeah, as I mentioned here https://github.com/cosmos/ibc-go/issues/6496#issuecomment-2191113393, I'm not too happy about it.

Changing it forces a migration on existing contracts. But I don't see any clear path forward on when you could switch over and fully deprecate the `Path` field. In fact, I think the longer it stays like this the more disruption it would cause to client contract users, as right now I don't think there is a large amount deployed and in use in prod, whereas if this continues to exist and more are deployed then we are forcing a larger amount of downtime in the event that we actually do change it at some point. 
Alternatively, it stays like this forever!